pkg/drivers/kic/oci/oci_test.go

```diff
@@ -130,6 +130,7 @@ func TestDaemonHost(t *testing.T) {
 		{"docker", "", "unix:///var/run/something", "127.0.0.1", false},
 		{"docker", "", "tcp://127.0.0.1/foo", "127.0.0.1", true},
 		{"docker", "", "ssh://127.0.0.1/bar", "127.0.0.1", true},
+		{"docker", "", "tcp://:", "127.0.0.1", true},
```

Comment:
consider adding more unit tests from the upstream that you pointed it out
https://github.com/moby/moby/blob/1f096174d614276845527790e556a6a3911ae99c/opts/hosts_test.go#L26

Reply:
@medyagh acknowledged but won't be until mid this week until I can get back to.

Reply:
@medyagh I want to ensure I understand the intent of the code here and have a few clarifying questions around DaemonHost and IsExternalDaemonHost as I add tests 

When a docker host is set to a tcp://127.0.0.1 the current implementation considers that remote as `u.Hostname() != ""`, which on re-reading confused me. However thinking on it and comparing to ssh://127.0.0.1:1234 which may ssh onto a VM that is exposed locally. I'm fine with following that as is for tcp 127.0.0.1 (with default port) is considered remote particularly as it could be port forwarded 

Should all tcp and ssh URLs be considered remote?

I also have a question how compatible with the DOCKER_HOST or podman CONTAINER_HOST handling does the oci driver handling need to be here?

Doing this would probably require more specific handling here of supported url Scheme -  podman supports `ssh`, `unix`, and `tcp`, and `npipe` on windows. Docker cli supports the same.

For docker it treats DOCKER_HOST where no scheme is present (identified via ://)  docker defaults to `tcp://` ie should minikube handle DOCKER_HOST=1.1.1.1:2222 (url.Parse will raise an error here). 

https://github.com/docker/cli/blob/a6351d0cd3db3ff540fed4e50d7c35339c1742a0/opts/hosts.go#L75

Docker also supports `fd://` for dockerd that only makes sense running under systemd, generally for dockerd using `docker.socket` and probably doesn't make sense for minikube to support that but wanted to check that also

```go
{"docker", "", "tcp://127.0.0.1/foo", "127.0.0.1", true},
```

Reply:
for podman we should do the same ! so lets change the line 695 too ! I believe both podman and docker should act the same, feel free to test to verify too. podman is free to install